complex/src/lib.rs

```diff
@@ -763,6 +900,36 @@ impl<T> serde::Deserialize for Complex<T> where
     }
 }
 
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub struct ParseComplexError {
+    kind: ComplexErrorKind,
+}
+
+#[derive(Copy, Clone, Debug, PartialEq)]
+enum ComplexErrorKind {
```

Comment:
I think this should be parameterized on `T: FromStr`, and have a variant containing `T::Err`.  If you can also have more specific variants for complex's own errors, that would be better.